src/Config.js

```diff
@@ -78,6 +78,55 @@ module.exports = class extends ChainedMap {
     }));
   }
 
+  toString({
+    verbose = false,
+    configPrefix = 'config'
+  } = {}) {
+    const stringify = require('javascript-stringify');
+
+    const config = this.toConfig();
+
+    return stringify(config, (value, indent, stringify) => {
+      // improve plugin output
+      if (value && value.__pluginName) {
+        const prefix = `/* ${configPrefix}.plugin('${value.__pluginName}') */\n`;
+        const constructorName = value.__pluginConstructorName;
+
+        if (constructorName) {
+          // get correct indentation for args by stringifying the args array and
+          // discarding the square brackets.
+          const args = stringify(value.__pluginArgs).slice(1, -1);
+          return prefix + `new ${constructorName}(${args})`;
+        } else {
+          return prefix + stringify({ args: value.__pluginArgs || [] });
```

Comment:
Are there know cases where this path is taken, or is it for defensive purposes?

Reply:
Mostly defensive when I found out about the case of copy-webpack-plugin, but this is probably unnecessary as a plugin has to be new-able and thus must have a name.